### PR TITLE
Move authkeys to config file

### PR DESCRIPTION
**Pull request by lclc**
From issue https://github.com/joushou/sshmuxd/issues/8

Do you want to have the same for the hostkey file?

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -21,43 +21,32 @@ type Host struct {
...
-		}
-
-		var (
-			pk      ssh.PublicKey
-			comment string
-		)
-
-		pk, comment, _, authFile, err = ssh.ParseAuthorizedKey(authFile)
+	for _, u := range us {
+		pk, _, _, _, err := ssh.ParseAuthorizedKey([]byte(u.Type + " " + u.PublicKey))
```
Comment (joushou) [inline main.go]: I think you can just use [`ssh.ParsePublicKey`](https://godoc.org/golang.org/x/crypto/ssh#ParsePublicKey) here. [`ssh.ParseAuthorizedKeys`](https://godoc.org/golang.org/x/crypto/ssh#ParseAuthorizedKey) just extract option, comments and type, and give the remainder to [`ssh.ParsePublicKey`](https://godoc.org/golang.org/x/crypto/ssh#ParsePublicKey)... If I read the code correctly, that is. The parsing logic in there is not super pretty.

File: README.md [hunk 2]
```diff
@@ -99,10 +98,24 @@ The format of the file is as follows (note that, due to the presence of comments
...
 	"hostkey": "hostkey",
 
-	// Authorized keys to use for authenticating users. An important note
-	// is that the comment (the part after the key itself in an entry)
-	// will	be used as name for the user internally.
-	"authkeys": "authkeys",
+	// Authorized keys to use for authenticating users.
+	"users": [
+		{
+			"type": "ssh-ed25519",
```
Comment (joushou) [inline README.md]: I believe this field is unnecessary, as the PEM blob itself contains the key type. We could leave the user list as simply a user/publicKey key/value pair, but then again, we might want per-user options later...

Maybe kill the type field, but leave the users as objects?

**Comment by joushou** [comment]
Sure, why not? Might as well join it all together while we're at it. It's just as single value, after all.

**Comment by lclc** [comment]
Thanks, makes sense. Updated.

Still need to move the hostkey, somehow it's not working yet because reading it from viper the SSH PEM formatted key seems to be invalid.

**Comment by joushou** [comment]
I merged it, but I forcibly removed it from master after noticing that you had accidentally added what seems to be a private configuration file. I missed it in the review as I was just looking for the change update.

Unfortunately, I don't know to clean up this pull request other than making a new one. If you're lucky, rewriting lclc:master will remove the information, but you should consider that host key compromised.

**Pull request merged**